packages/core/src/utils/service-initializer.ts

```diff
@@ -121,14 +123,31 @@ export async function createAgentServices(
     await pluginManager.initialize(config.plugins.custom);
     logger.info('Plugin manager initialized');
 
-    // 7. Initialize tool manager with internal tools options
-    // 7.1 - Create allowed tools provider based on configuration
+    // 7. Initialize FileSystemService and ProcessService for internal tools
+    const fileSystemService = new FileSystemService({
+        allowedPaths: ['.'],
+        blockedPaths: ['.git', 'node_modules/.bin', '.env'],
```

Comment:
we should also block parent paths outside of the curr directory (../) for safety - not sure if we are doing this rn

edit: nvm saw we are handling this in path-validator.ts